src/geckomat/gather_kcats/runDLKcatFromDocker.m

```diff
@@ -0,0 +1,111 @@
+    DLKcatPath = fullfile(geckoPath,'dlkcat');
+end
+
+if nargin < 3 || isempty(modelAdapter)
+    modelAdapter = ModelAdapterManager.getDefaultAdapter();
+    if isempty(modelAdapter)
+        error('Either send in a modelAdapter or set the default model adapter in the ModelAdapterManager.')
+    end
+end
+params = modelAdapter.params;
+
+if nargin < 2 || isempty(DLKcatFile)
+    DLKcatFile = fullfile(params.path,'data','DLKcat.tsv');
+end
+
+if nargin < 1 || isempty(DLKcatPath)
+    deleteImage = false;
+end
+
+%% Check and install requirements
+% On Mac, docker might not be properly loaded if MATLAB is started via
+% launcher and not terminal.
+if ismac
+    setenv('PATH', strcat('/usr/local/bin', ':', getenv("PATH")));
+end
+
+% Check if docker is installed
+[checks.docker.status, checks.docker.out] = system('docker --version');
+if checks.docker.status ~= 0
+    error('Cannot find Docker. Make sure it is installed')
+end
+
+% Dowload DLKcat package
+if ~exist(fullfile(DLKcatPath,'DLKcat.py'),'file')
+    if ~exist(fullfile(DLKcatPath),'dir')
+        mkdir(fullfile(DLKcatPath));
+    end
+    disp('=== Downloading DLKcat ...')
+    packageURL = 'https://github.com/SysBioChalmers/GECKO/raw/dlkcatPackage/dlkcat.zip';
+    %packageURL = 'https://github.com/SysBioChalmers/GECKO/releases/download/v3.0.0/dlkcat_package.zip';
+    websave(fullfile(DLKcatPath,'dlkcat_package.zip'),packageURL);
+    unzip(fullfile(DLKcatPath,'dlkcat_package.zip'),geckoPath);
+    delete(fullfile(DLKcatPath,'dlkcat_package.zip'));
+end
+
+currPath = pwd();
+cd(DLKcatPath);
+
+% First check if the Docker image exist
+[checks.image.status, checks.image.out] = system('docker images');
+if checks.image.status == 0 && ~contains(checks.image.out,'dlkcat')
+    % Check if the image file is already dowloaded
+    if ~exist(fullfile(DLKcatPath,'dlkcat_docker.tar.gz'),'file')
+        disp('=== Downloading DLKcat docker image, this may take several minutes...')
+        packageURL = 'https://github.com/SysBioChalmers/GECKO/raw/dlkcatPackage/dlkcat_docker.tar.gz';
+        websave(fullfile(DLKcatPath,'dlkcat_docker.tar.gz'),packageURL);
+    end
+    [checks.dockerload.status, checks.dockerload.out] = system('docker load --input dlkcat_docker.tar.gz');
+    if deleteImage
+        delete(fullfile(DLKcatPath,'dlkcat_docker.tar.gz'));
+    end
+    if checks.dockerload.status ~= 0
+        error('Fail to load the image')
+    end
+elseif checks.image.status ~= 0
+    error('Docker is installed. Check if the docker desktop app is running')
+end
+
+% Copy the input file for needed for the docker image to run DLKcat
+copyfile(DLKcatFile,fullfile(DLKcatPath, 'data/DLKcat.tsv'));
+
+disp('=== Running DLKcat prediction, this may take several minutes...')
+% In the next line, pythonPath does not need to be specified, because it is
+% already mentioned when building the virtualenv.
+dlkcat.status = system('docker compose -f docker-compose.yaml up dlkcat-cont', '-echo');
```

Comment:
Based on how I've used Docker images before, I have a feeling we could do without `docker-compose`, and use instead `docker run`. To be sure, I wanted to look into the `docker-compose.yaml` file, but I couldn't seem to find it.

Reply:
Yo can do it, but the problem is that runnnig DLKCat needs a few files that defined at the moment that is running. 

Compose file is in the DLKcat folder

Reply:
GECKO on branch `feat/runDLKcatFromDocker `
❯ find . -name "*.yaml"

yields no results

@ae-tafur could you please point me directly to the file?

Reply:
> needs a few files defined at the moment that is running.

This is routinely done with [`bind` mounts](https://docs.docker.com/storage/bind-mounts/#start-a-container-with-a-bind-mount), so it might be possible to not have a `docker-compose` file.

Reply:
https://github.com/SysBioChalmers/GECKO/blob/2a9b890cd0495ff9a532a165e3b00ba6a875ec21/dlkcat.zip